extensions/db/src/main/java/org/mule/extension/db/internal/domain/type/MetadataDbTypeManager.java

```diff
@@ -83,10 +83,12 @@ protected void initialise(DbConnection connection) {
       while (resultSetIterator.hasNext()) {
         Map<String, Object> typeRecord = resultSetIterator.next();
 
-        Number data_type = (Number) typeRecord.get(METADATA_TYPE_ID_COLUMN);
-        String type_name = (String) typeRecord.get(METADATA_TYPE_NAME_COLUMN);
+        Number dataType = (Number) typeRecord.get(METADATA_TYPE_ID_COLUMN);
+        String typeName = (String) typeRecord.get(METADATA_TYPE_NAME_COLUMN);
 
-        ResolvedDbType resolvedDbType = new ResolvedDbType(data_type.intValue(), type_name);
+        DbType resolvedDbType = dataType.equals(Types.BLOB)
```

Comment:
There is something wrong here, because this code is not taking in account other types (like struct and array), same problem exists on 3.x :(